src/Api/Auth/Controllers/WebAuthnController.cs

```diff
@@ -93,6 +114,29 @@ private async Task ValidateRequireSsoPolicyDisabledOrNotApplicable(Guid userId)
         }
     }
 
+    [HttpPut()]
+    public async Task UpdateCredential([FromBody] WebAuthnLoginCredentialUpdateRequestModel model)
+    {
+        var tokenable = _assertionOptionsDataProtector.Unprotect(model.Token);
+        if (!tokenable.TokenIsValid(WebAuthnLoginAssertionOptionsScope.PrfRegistration))
+        {
+            throw new BadRequestException("The token associated with your request is expired. A valid token is required to continue.");
+        }
+
+        var (_, credential) = await _userService.CompleteWebAuthLoginAssertionAsync(tokenable.Options, model.DeviceResponse);
+        if (credential == null)
+        {
+            throw new BadRequestException("Unable to update WebAuthnLoginCredential.");
```

Comment:
nit: maybe make this a bit more user friendly since this will probably be shown to the user

```suggestion
            throw new BadRequestException("Unable to update credential.");
```